library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -992,20 +994,25 @@ impl<T, A: Allocator> VecDeque<T, A> {
         // * The head of the VecDeque is moved before calling `drop_in_place`,
         //   so no value is dropped twice if `drop_in_place` panics
         unsafe {
-            if len > self.len() {
+            if len >= self.len {
                 return;
             }
-            let num_dropped = self.len() - len;
+
+            if !mem::needs_drop::<T>() {
```

Comment:
same as my other comment; no need for `needs_drop` checks here -- the code below will easily optimize to just `self.len = len;` since the `drop_in_place` does nothing and thus all the other calculations are unused.